File: src/main/java/my_salary/classes/WeekEndInfo.java
package my_salary.classes;

import lombok.Data;

import java.time.DayOfWeek;
import java.time.OffsetDateTime;

@Data
public class WeekEndInfo {
    public Float weekEndStartHour;
    public DayOfWeek weekEndStartDay;
    public Float weekEndEndHour;
    public DayOfWeek weekEndEndDay;
    public Float weekEndPercentage;

    @Override
    public String toString() {
        return "WeekEndInfo{" + "weekEndStartHour=" + weekEndStartHour + ", weekEndStartDay='" + weekEndStartDay + '\'' + ", weekEndEndHour=" + weekEndEndHour + ", weekEndEndDay='" + weekEndEndDay + '\'' + ", weekEndPercentage=" + weekEndPercentage + '}';
    }
}

File: client/lib/Model/WeekEndInfo.dart
import 'package:intl/intl.dart';

class WeekEndInfo {
  DateTime? weekEndStartHour;
  DateTime? weekEndEndHour;
  double? weekEndPercentage;
  String? weekEndStartDay;
  String? weekEndEndDay;

  WeekEndInfo({
    this.weekEndStartHour,
    this.weekEndEndHour,
    this.weekEndPercentage,
    this.weekEndStartDay,
    this.weekEndEndDay,
  });

  factory WeekEndInfo.fromJson(Map<String, dynamic> json) {
    DateFormat format = DateFormat("HH:mm:ss");
    return WeekEndInfo(
      weekEndStartHour: format.parse(json['weekEndStartHour']),
      weekEndEndHour: format.parse(json['weekEndEndHour']),
      weekEndPercentage: (json['weekEndPercentage'] as num?)?.toDouble(),
      weekEndStartDay: json['weekEndStartDay'],
      weekEndEndDay: json['weekEndEndDay'],
    );
  }

  Map<String, dynamic> toJson() {
    DateFormat format = DateFormat("HH:mm:ss");
    return {
      'weekEndStartHour': format.format(weekEndStartHour!),
      'weekEndEndHour': format.format(weekEndEndHour!),
      'weekEndPercentage': weekEndPercentage,
      'weekEndStartDay': weekEndStartDay,
      'weekEndEndDay': weekEndEndDay,
    };
  }
}